### DISCUSSION TITLE
[Linear solver stopped? or Unfinised]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#17883] [Q&A General] [answered]
Dear moose users,

I am currently trying to simulate my numerical model and find a reason for unfinished calculations.

My problem is like the below picture.

In a certain time step, it seems like linear residual calculation does not end.

It just stuck in there like the below. (Timestep 2200...)

I am trying to figure the reason for over two weeks but I can't find any clue.

So, if you had a similar problem or solution please let me know.

Best regards,
Q
![image](https://user-images.githubusercontent.com/35205601/118839794-c5d35780-b901-11eb-853c-75d2f32bc54e.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

At 10^-17 your solve is very much converged for that timestep. You should set nl_rel_tol and nl_abs_tol so that the code does not try to converge that further. Numerical floating point precision is not that far. 

Could you please print the linear solve history, see [SolutionHistory](https://mooseframework.inl.gov/source/outputs/SolutionHistory.html).
The default is max 10,000 linear iterations, which can take a while if convergence is not reached.

Guillaume

**Reply by coskrrb2002** [reply]
Dear GiudGiud,


Thank you for your comment.

This morning, I finally found the error and fixed it!

The error was originated from my material kernel which takes a lot of time to calculate...

But your comment will be helped me a lot!


Sincerely,
Q
